packages/engine/lib/error/src/single_provider.rs

```diff
@@ -0,0 +1,24 @@
+use core::fmt;
+
+use crate::provider::{Demand, Provider};
+
+/// Wrapper-structure to provide `T`.
+pub struct SingleProvider<T>(pub T);
```

Comment:
I wonder if we want to be a little clearer about the intention of this and why it's kind of a special case.

Perhaps we should call this `GeneralSingleProvider` or something, or at least change the doc-comment.

Reply:
I don't think `wrap` is encapsulating the information here. The point of this Provider is to apply a **general** mechanism to **provide** _any_ **single** thing.